pkg/virt-controller/network/pod_annotations_test.go

```diff
@@ -0,0 +1,112 @@
+package network_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	virtv1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/deviceinfo"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/network"
+)
+
+var _ = Describe("pod annotations", func() {
+	const (
+		booHashedIfaceName = "pod6446d58d6df"
+		fooHashedIfaceName = "pod2c26b46b68f"
+	)
+
+	networkStatus := `
+[
+{
+  "name": "default/no-device-info",
+  "interface": "pod6446d58d6df",
+  "mac": "8a:37:d9:e7:0f:18",
+  "dns": {}
+},
+{
+  "name": "default/with-device-info",
+  "interface": "pod2c26b46b68f",
+  "dns": {},
+  "device-info": {
+    "type": "pci",
+    "version": "1.0.0",
+    "pci": {
+      "pci-address": "0000:65:00.2"
+    }
+  }
+}
+]`
+
+	Context("Generate pod network annotations", func() {
+		It("should have empty device info annotation when there are no networks", func() {
```

Comment:
Please consider using the "equal" matcher instead of (`haveLen` + `haveKeyWithValue`), it can provide better visibility for expected results.

In general it can be easier to maintain: in case there it require adding another annotation, you could change the expected annotation slice instead adding new assert and incrementing the haveLen assert.